lib/extensions/ar_migration.rb

```diff
@@ -0,0 +1,84 @@
+module ArPglogicalMigration
+  module PglogicalMigrationHelper
+    def self.migrations_column_present?
+      ActiveRecord::Base.connection.columns("miq_regions").any? { |c| c.name == "migrations_ran" }
+    end
+
+    def self.update_local_migrations_ran(version, direction)
+      return unless migrations_column_present?
+      return unless (region = MiqRegion.my_region)
+
+      new_migrations = ActiveRecord::SchemaMigration.normalized_versions
+      new_migrations << version if direction == :up
+      migrations_value = ActiveRecord::Base.connection.quote(PG::TextEncoder::Array.new.encode(new_migrations))
+
+      ActiveRecord::Base.connection.exec_query(<<~SQL)
+        UPDATE miq_regions
+        SET migrations_ran = #{migrations_value}
+        WHERE id = #{region.id}
+      SQL
+    end
+  end
+
+  class RemoteRegionMigrationWatcher
+    class HelperARClass < ActiveRecord::Base; end
+
+    attr_reader :subscription, :version, :region
+
+    def initialize(subscription, version)
+      @subscription = subscription
+      @version = version
+      @region = MiqRegion.find_by(:region => subscription.provider_region)
+    end
+
+    def wait_for_remote_region_migration(wait_time = 1)
+      return unless wait_for_migration?
+
+      Vmdb.rails_logger.info(wait_message)
```

Comment:
Any reason to use this rather than `Rails.logger`?

Reply:
I figured that `Vmdb` logger was the better one since it goes through the whole multicast thing, but I can change it if we should prefer `Rails.logger`. Or I could also use `$rails_log` (which is what `Vmdb.rails_logger` is.

Reply:
> since it goes through the whole multicast thing

Okay, that's probably better.  I was just trying to divorce this extension from manageiq as much as possible